pkgs/build-support/trivial-builders/test/default.nix

```diff
@@ -33,4 +33,5 @@ recurseIntoAttrs {
     inherit (references) samples;
   };
   writeTextFile = callPackage ./write-text-file.nix {};
+  addTmateBreakpoint = callPackage ./addTmateBreakpoint.nix {};
```

Comment:
I'm scared of running this test on Hydra / ofBorg / nix.ci. Is there a way to prove that this is working that _doesn't_ mean there could be an open connection out?

Reply:
We could re-override the attrs to make it no longer a FOD, and maybe look for some log lines, and grep them to ensure that `tmate` was spawned. I'm not sure how to do this with the trivial-builders testing infrastructure though, as I don't think there's a way of doing "golden testing", i.e checking output matches an expected string. I'm not sure how to best implement this, suggestions definitely welcome.

Reply:
I wonder if you could use the nixosTests infrastructure instead, and do the Nix build on a VM which was guaranteed not to have network connectivity out... then scrape the build output for the whole nixosTest derivation using something like [`testBuildFailure`](https://nixos.org/manual/nixpkgs/unstable/#tester-testBuildFailure) to see the appropriate incantations are added.

Even better would look like a second VM which connected to the first through the SSH connection to prove the end-to-end workflow was successful. We have a couple of those in the nixosTest world, especially in the networking tests.

Reply:
I'm not sure I have the kind of time needed to implement that kind of test, and the value is unknown since tmate is not self-hosted anyway. A self-hosted tmate would be possible with the test you suggested which would be end-to-end. But yeah, I don't have the time to implement that.

Reply:
Yeah, that's super fair. I don't think I'm comfortable with having this specific test in CI (as I currently understand it) so this might need to go into the "documented but not tested" bucket.

I'm also open to learning about CI and why this is safe.